app/src/main/java/com/example/dragonx/presentation/RocketList/RocketListParser.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.dragonx.presentation.RocketList
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.dragonx.NetworkService.NetworkService
+import com.example.dragonx.NetworkService.Rocket
+import com.example.dragonx.NetworkService.Rockets
+import com.example.dragonx.util.RocketModel
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+enum class ApiStatus {LOADING, ERROR, DONE}
+
+class RocketListParser {
+    private val _status = MutableLiveData<ApiStatus>()
+    val status: LiveData<ApiStatus>
+        get() = _status
+
+    private suspend fun parseJson() : List<Rocket>? {
+        try {
+        _status.postValue(ApiStatus.LOADING)
+        val rocketsList: List<Rocket> = NetworkService.getInstance()
```

Comment:
Read about dependency injection please, you need to think how to split using instance from it creation.